lightningd/channel/channel.c

```diff
@@ -1339,6 +1364,18 @@ static struct io_plan *req_in(struct io_conn *conn, struct daemon_conn *master)
 	}
 
 out:
+	/* In case we've now processed reply, process packet backlog. */
+	if (!peer->handle_master_reply) {
+		const u8 *msg = msg_dequeue(&peer->master_deferred);
+		if (msg) {
+			/* Free old packet exactly like daemon_conn_read_next */
+			master->msg_in = tal_free(master->msg_in);
+			master->msg_in = cast_const(u8 *, tal_steal(master,msg));
+			return req_in(conn, master);
```

Comment:
Doing a loop would have meant more of a refactor; this is a bit weird (true) but it ensures we handle them exactly the same.